helion/_compiler/device_ir.py

```diff
@@ -746,6 +747,45 @@ def visit_Subscript(self, node: ast.Subscript) -> object:
         return hl.subscript(self.visit(value), self._subscript_slice_proxy(node.slice))
 
     def visit_Call(self, node: ast.Call) -> object:
+        # Handle Python builtin print call
+        if (
+            isinstance(node.func, ast.Name)
+            and node.func._type_info.value is builtins.print  # pyre-ignore[16]
+        ):
+            # Convert print to hl.device_print
+            args = []
+            for arg in node.args:
+                if isinstance(arg, ast.Starred):
+                    args.extend(self._to_proxy(arg.value))  # pyre-ignore[6]
+                else:
+                    args.append(self._to_proxy(arg))
+
+            # Check that we have at least one argument (prefix)
+            if len(args) == 0:
+                raise ValueError("print() requires at least one argument (prefix)")
+
+            # First argument must be the prefix string
+            if not isinstance(args[0], str):
+                raise TypeError(
+                    f"First argument to print() must be a string prefix, got {type(args[0])}"
+                )
+
+            prefix = args[0]
+            tensor_args = args[1:]
+
+            # For compile-time values like tensor shapes, we should error out
+            for i, arg in enumerate(tensor_args):
+                if not isinstance(arg, torch.Tensor):
+                    raise TypeError(
+                        f"print() only supports runtime tensor values. "
+                        f"Argument {i + 1} is {type(arg).__name__}, not a tensor. "
+                        f"Compile-time values like tensor shapes are not supported yet."
+                    )
```

Comment:
This error checking should happen at type checking time and should go in the meta function for `hl.device_print()`.

If you put it here, then I can bypass the error checking by calling `hl.device_print` directly.